src/control/system/properties.go

```diff
@@ -40,3 +46,77 @@ func GetMgmtProperty(db SysAttrGetter, key string) (string, error) {
 func DelMgmtProperty(db SysAttrSetter, key string) error {
 	return SetMgmtProperty(db, key, "")
 }
+
+// SetUserProperties sets the user-visible properties based on the supplied key/value pairs.
+func SetUserProperties(db SysAttrSetter, sysProps daos.SystemPropertyMap, toSet map[string]string) error {
```

Comment:
we are setting system properties but using the system attribute setter in the db, is there a clear distinction between system attributes and properties and if so what is the definition? in the database is that the same distinction or are they treated /stored both as attributes?

Reply:
The underlying storage for both properties and attributes is a single `map[string]string` in the raft-backed system database. The distinction between attributes and properties is that attributes are free-form key/value pairs set by the admin, and properties have a defined set of keys and expected values. I didn't see value in making it more complicated for now.